iron-list.html

```diff
@@ -589,7 +637,12 @@
     updateViewportBoundaries: function() {
       var scrollerStyle = window.getComputedStyle(this._scroller);
       this._scrollerPaddingTop = parseInt(scrollerStyle['padding-top'], 10);
-      this._viewportSize = this._scroller.offsetHeight;
+      this._viewportHeight = this._scroller.offsetHeight;
+
+      // If in grid mode compute the _viewportWidth and _physicalItemsPerRow
+      if (this.grid) {
+        this._updateGridMetrics();

```

Comment:
My thought process behind having `_updateGridMetrics` broken out into its own function was that, to me, any update to `_viewportWidth` should also result in the recalculation of `_physicalItemsPerRow`. 

Because I needed to update both `_viewportHeight` and `_viewportWidth` inside of `updateViewportBoundaries` I choose the separate function route.